frontend/src/host_orchestrator/orchestrator/controller_test.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package orchestrator
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	apiv1 "github.com/google/android-cuttlefish/frontend/src/liboperator/api/v1"
+
+	"github.com/gorilla/mux"
+)
+
+type testIM struct{}
+
+func (m *testIM) CreateCVD(req apiv1.CreateCVDRequest) (Operation, error) {
+	return Operation{}, nil
+}
+
+func TestCreateCVDSucceeds(t *testing.T) {
+	rr := httptest.NewRecorder()
+	router := mux.NewRouter()
+	req, err := http.NewRequest("POST", "/cvds", strings.NewReader("{}"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	controller := Controller{InstanceManager: &testIM{}}
+	controller.AddRoutes(router)
+
+	router.ServeHTTP(rr, req)
+
+	expected := http.StatusOK
+	if rr.Code != expected {
+		t.Errorf("unexpected status code <<%d>>, want: %d", rr.Code, expected)
+	}
```

Comment:
This test is almost the same as the one below and you're likely to add similar tests in the future. I suggest you refactor like this:
```
func requestFromTestRouter(t *testing.T, r *http.Request, expected http.StatusCode) {
	rr := httptest.NewRecorder()
	router := buildTestRouter()
	
        router := mux.NewRouter()
        controller := Controller{InstanceManager: &testIM{}}
	controller.AddRoutes(router)

	if rr.Code != expected {
		t.Errorf("unexpected status code <<%d>>, want: %d", rr.Code, expected)
	}
}

func TestCreateCVDSucceeds(t *testing.T) {
	req, err := http.NewRequest("POST", "/cvds", strings.NewReader("{}"))
	if err != nil {
		t.Fatal(err)
	}
        requestFromTestRouter(t, router, req, http.StatusOk)
}

func TestGetOperationSucceeds(t *testing.T) {
	req, err := http.NewRequest("GET", "/operations/"+op.Name, nil)
	if err != nil {
		t.Fatal(err)
	}
        requestFromTestRouter(t, router, req, http.StatusOk)
}
```

Reply:
with this refactoring you're always using the same Controller instance and each test should create its own controller instance to test its specific behavior.

I agree there's room for refactoring, so I created a helper method to make the requests which do not obscure the relationship between arrangement and assertions.

Reply:
Actually, each test would create a new controller instance, it would just happen to be created in the same way.

With the proposed refactor each controller instance would have an actual (test) object on each of its attributes, which is arguably better than having `nil` on some, so I think it's a net improvement.

Reply:
the proposed `requestFromTestRouter` helper method obscures how the controller is created which is what we are unit testing, this is a problem for writing new tests that requires a change in the controller, i.e a new field is added and needs to be set, in such cases you'll need to search for, understand and modify `requestFromTestRouter` which is not ideal, this also affects the existing tests.

Reply:
SG